enterprise/api/v1/views.py

```diff
@@ -64,6 +64,71 @@ class EnterpriseCustomerViewSet(EnterpriseReadOnlyModelViewSet):
+        authentication_classes=[JwtAuthentication, BearerAuthentication, SessionAuthentication],
+        throttle_classes=[ServiceUserThrottle],
+    )  # pylint: disable=invalid-name,unused-argument
+    def courses(self, request, pk=None):
+        """
+        Retrieve the list of courses contained within the catalog linked to this enterprise.
+
+        Only courses with active course runs are returned. A course run is considered active if it is currently
+        open for enrollment, or will open in the future.
+        ---
+        serializer: serializers.CourseSerializerExcludingClosedRuns
+        """
+        enterprise_customer = self.get_object()
+
+        # Make sure there is a catalog associated with the enterprise
+        if not enterprise_customer.catalog:
+            error_message = (
+                "No catalog is associated with Enterprise {enterprise_name} from endpoint '{path}'.".format(
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise NotFound("The resource you are looking for does not exist: " + error_message)
+
+        # Ensure that the user making this request is either a staff user or is linked to the enterprise
+        if not request.user.is_staff and not models.EnterpriseCustomerUser.objects.filter(
+                enterprise_customer=enterprise_customer,
+                user_id=request.user.id,
+        ).exists():
+            error_message = (
+                "User '{username}' is not associated with Enterprise {enterprise_name} from endpoint '{path}'.".format(
+                    username=request.user.username,
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise PermissionDenied("The user does not have permission to access this resource: " + error_message)
+
+        # We have handled potential error cases and are now ready to call out to the Catalog API.
+        catalog_api = CourseCatalogApiClient(request.user)
+        courses = catalog_api.get_paginated_catalog_courses(enterprise_customer.catalog, request.GET)
+
+        # An empty response means that there was a problem fetching data from Catalog API, since
+        # a Catalog with no courses has a non empty response indicating that there are no courses.
+        if not courses:
+            error_message = (
+                "Unable to fetch API response for catalog courses for Enterprise {enterprise_name} from endpoint "
+                "'{path}'.".format(
+                    enterprise_name=enterprise_customer.name,
+                    path=request.get_full_path()
+                )
+            )
+            logger.error(error_message)
+            raise NotFound("The resource you are looking for does not exist: " + error_message)
```

Comment:
It might be better for the message to indicate that there was a problem with retrieving the catalog instead of saying that the resource doesn't exist.